src/main/java/org/apache/ibatis/executor/resultset/DefaultResultSetHandler.java

```diff
@@ -697,15 +698,24 @@ private boolean allowedConstructor(final Constructor<?> constructor, final List<
     if (parameterTypes.length != classNames.size()) return false;
     for (int i = 0; i < parameterTypes.length; i++) {
       final Class<?> parameterType = parameterTypes[i];
-      if (parameterType.isPrimitive() && !primitiveTypes.getWrapper(parameterType).getName().equals(classNames.get(i))) {
+      final String className = classNames.get(i);
+      if (parameterType.isPrimitive() && !primitiveTypes.getWrapper(parameterType).getName().equals(className)) {
         return false;
-      } else if (!parameterType.isPrimitive() && !parameterType.getName().equals(classNames.get(i))) {
+      } else if (!parameterType.isPrimitive() && !parameterType.isAssignableFrom(toClassForName(className))) {
```

Comment:
Good change! I should of supported that in the primitiveTypes wrapper logic since it's really wrapper types.

Reply:
Not a big deal, but this part is a little bit hard to read due to negative conditions.
It can be simplified like this...
```java
return parameterType.isPrimitive() ?
  primitiveTypes.getWrapper(parameterType).getName().equals(className) :
  parameterType.isAssignableFrom(toClassForName(className));
```

...or this with a minor refactoring.

```java
return parameterType.isPrimitive() ?
  primitiveTypes.isWrapper(parameterType, className) :
  parameterType.isAssignableFrom(toClassForName(className));
```

Sorry if I missed something!